azurerm/internal/services/machinelearning/resource_arm_machine_learning_workspace.go

```diff
@@ -0,0 +1,432 @@
+		return fmt.Errorf("Error waiting for creation of Machine Learning Workspace %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Machine Learning Workspace %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Machine Learning Workspace %q (Resource Group %q) ID", name, resGroup)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmMachineLearningWorkspaceRead(d, meta)
+}
+
+func resourceArmMachineLearningWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.WorkspaceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing Machine Learning Workspace ID `%q`: %+v", d.Id(), err)
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku_name", sku.Name)
+	}
+
+	if props := resp.WorkspaceProperties; props != nil {
+		d.Set("application_insights_id", props.ApplicationInsights)
+		d.Set("storage_account_id", props.StorageAccount)
+		d.Set("key_vault_id", props.KeyVault)
+		d.Set("container_registry_id", props.ContainerRegistry)
+		d.Set("description", props.Description)
+		d.Set("friendly_name", props.FriendlyName)
+	}
+
+	if err := d.Set("identity", flattenArmMachineLearningWorkspaceIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("Error flattening identity on Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmMachineLearningWorkspaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.WorkspacesClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.WorkspaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	update := machinelearningservices.WorkspaceUpdateParameters{
+		WorkspacePropertiesUpdateParameters: &machinelearningservices.WorkspacePropertiesUpdateParameters{},
+	}
+
+	if d.HasChange("sku_name") {
+		skuName := d.Get("sku_name").(string)
+		update.Sku = &machinelearningservices.Sku{
+			Name: &skuName,
+		}
+	}
+
+	if d.HasChange("description") {
+		description := d.Get("description").(string)
+		update.WorkspacePropertiesUpdateParameters.Description = &description
```

Comment:
_minor_ these could be
```suggestion
		update.WorkspacePropertiesUpdateParameters.Description = &utils.String(d.Get("description").(string))
```

Reply:
Done, and I also changed some other field assignment with similar pattern